### DISCUSSION TITLE
Can't see images in shared documents

### DISCUSSION THREAD

**Discussion by syphax-bouazzouni** [outline/outline#4649] [Self Hosting]
Hi,

I self-hosted an outline instance, using the docker-compose file, with Minio for the S3 storage.

## The issue

Everything works fine when logged in,  we can upload files/images and access them. But when we share the document for a no logged users we have no access to those files

For example this document: https://doc.jonquetlab.lirmm.fr/share/61e010bd-0a7c-4806-bcff-b67a17414c4b

### What we see if logged in  
![image](https://user-images.githubusercontent.com/29259906/210800461-c7d058aa-296c-42bf-abcd-64c701e9d71e.png)

### And when not logged in 
<img width="1777" alt="image" src="https://user-images.githubusercontent.com/29259906/210800663-ed34c15b-a424-4200-86d0-4911325f34f5.png">

## The error behind

when we are logged in to access the image, we call this https://doc.jonquetlab.lirmm.fr/api/attachments.redirect?id=85fc252a-1ca0-4879-b98b-a810fac1a916

But when not logged in it's a direct call to minio: https://doc.jonquetlab.lirmm.fr/minio/storage/public/9b853603-c253-4504-9678-dc58e0fc50ff/1e6d0d23-349d-44fb-8769-6cd3ade300d6/1920px-Lod-cloud_2020-07-27.png?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=minioadmin%2F20230105%2Feu-central-1%2Fs3%2Faws4_request&X-Amz-Date=20230105T135826Z&X-Amz-Expires=60&X-Amz-Signature=69e05824135746337fdb40b811c4fa5f2d8574d7e5cd6004a391efb3c29e8b9d&X-Amz-SignedHeaders=host

And we get a **SignatureDoesNotMatch**
<img width="1777" alt="image" src="https://user-images.githubusercontent.com/29259906/210800934-14930e44-4f3d-4a01-a87b-e9a56c361166.png">

## My configuration

- Outline is running on localhost:3000 and reversed proxy to  https://doc.jonquetlab.lirmm.fr
- Minio is running on localhost:9000 and reversed proxy to https://doc.jonquetlab.lirmm.fr/minio
- Docker.env:
   ```
    AWS_ACCESS_KEY_ID=minioadmin
    AWS_SECRET_ACCESS_KEY=password
    AWS_REGION=eu-central-1
    AWS_S3_UPLOAD_BUCKET_URL=https://doc.jonquetlab.lirmm.fr/minio
   AWS_S3_UPLOAD_BUCKET_NAME=storage
   AWS_S3_UPLOAD_MAX_SIZE=26214400
   AWS_S3_FORCE_PATH_STYLE=true
   AWS_S3_ACL=public-read
   ```

**Comment by Limezy** [comment]
@syphax-bouazzouni clicking on your link, it seems that you have resolved your issue.
Would you mind explaining how you did ? Facing the exact same problem right now.

**Reply by syphax-bouazzouni** [reply]
My hacky solution was to configure our server proxy to remove the amazon request parameters 

```
 ProxyPass /minio/ http://localhost:9000/
 ProxyPassReverse /minio/ http://localhost:9000/
  
 RewriteEngine    On
 RewriteCond %{REQUEST_URI} .*/storage/public.*
 RewriteCond      %{QUERY_STRING}    (.*)X-Amz-Signature=(.*)
 RewriteRule      (.*)               $1?     [R=permanent]
```

**Reply by Limezy** [reply]
@tommoor is that normal ?
It seems to me that it is a bug, isn't it ? Thanks for your help.

@syphax-bouazzouni many thanks for your help. Do I understand well that this is your nginx configuration on MinIO side ?

**Reply by syphax-bouazzouni** [reply]
Yeah, my apache server configuration, which we use as a proxy for our MinIO server.

**Reply by tommoor** [reply]
I'm not aware of anyone else setting it up this way.

The storage setup is fairly simple, there is one bucket called "public" that contains images with public permissions, this can be safely made public in Minio. The rest are private. Every asset that's uploaded to storage is given the correct ACL permissions.

**Reply by Limezy** [reply]
@tommoor I understand that, and I have had a MinIO + Outline instance working perfectly for 10 months now. After investigating, my problem is slightly different : I can see images in the shared documents, and they have the correct permissions. However, I get a blank on on my logo image, as per the screenshot below. 

<img width="1552" alt="Capture d’écran 2023-02-03 à 22 37 02" src="https://user-images.githubusercontent.com/24638389/216644845-f93fa271-d10f-41f1-be7d-9d9aa69ae794.png">

When I try to reach that URL without being logged-in, I get a 401 error. When I try to reach that same URL while being logged-in, everything works perfectly. My whole bucket is set to public.

Same problem exactly for the logo on public shares (the images inside the documents are working perfectly)
<img width="1552" alt="Capture d’écran 2023-02-03 à 22 46 21" src="https://user-images.githubusercontent.com/24638389/216647755-02617380-5c2b-4506-ab4d-72234dd509d7.png">

<img width="1552" alt="Capture d’écran 2023-02-03 à 22 50 36" src="https://user-images.githubusercontent.com/24638389/216647877-edbc56f7-acab-412f-8a9b-22ef4d1e8c1e.png">

Before, that part was the hardcoded Outline logo, so I guess this is a new addition to the codebase that may have gone wrong for the self-hosted folks.
Or it something i'm doing wrong ? Thanks in advance for your kind help

**Reply by ovizii** [reply]
> I'm not aware of anyone else setting it up this way.
> 
> The storage setup is fairly simple, there is one bucket called "public" that contains images with public permissions, this can be safely made public in Minio. The rest are private. Every asset that's uploaded to storage is given the correct ACL permissions.

Do you really meant to say “one bucket called public”? I thought there was a bucket used for outline and within there is one folder named public which should be public?

**Reply by tommoor** [reply]
Yes, I meant folder sorry.

@Limezy could you try re-uploading your logo image in settings.

**Reply by Limezy** [reply]
@tommoor Oh, it works !

**Reply by ovizii** [reply]
> Yes, I meant folder sorry.
> 
> @Limezy could you try re-uploading your logo image in settings.

Thanks for clarifying. Let me add one more question, does the avatars folder need to be public too? 
Currently, in my minio bucket for outline I see 3 folders: avatars, public, uploads and so far I have only made the one named “public”, public.
